chaosaws/utils.py

```diff
@@ -0,0 +1,4 @@
+#
+def breakup_iterable(values: list, limit: int = 50) -> list:
```

Comment:
I wonder if one of the functions from the builtin itertools might have helped?

https://docs.python.org/3/library/itertools.html#module-itertools

Reply:
For instance:

```
from itertools import tee
for i in tee(iterable, 4):
    ...
```

?

just a thought

Reply:
i looked at tee, but i didn't find the usage very friendly or as effective.